[ROCm][Windows] Disable hipSPARSE and CK declarations and remove references for Windows

ikalinic opened the pull request:
This PR removes references to `hipSPARSE` and `ck` functions and disables declarations which are not supported on Windows.

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang @naromero77amd

m-gallus committed:
Disable HIP sparse declarations for Windows

Disables references to the HIP Sparse functions, since they are currently unsupported

(cherry picked from commit 047147ac0e0f3cd022a1f7bdab72c5c21a3d1cf7)

m-gallus committed:
Remove reference to a complex sparse hip blas function.

Also remove sparse reference that is currently unavailable.

(cherry picked from commit d745bfba8f9d91f9dd4fb92aa40e6c3671a3567c)

ikalinic committed:
Disable HIP sparse and CK declarations for Windows

Disables HIP sparse declarations and removes references for HIP sparse
and CK for Windows, since they are currently unsupported

ikalinic commented:
@pytorchbot label "topic: not user facing"

jeffdaily commented on aten/src/ATen/cuda/CUDABlas.cpp:
```diff
@@ -2124,7 +2124,7 @@ void vdot<c10::complex<double>>(CUDABLAS_DOT_ARGTYPES(c10::complex<double>)) {
                                    reinterpret_cast<cuDoubleComplex*>(result)));
 }
-
+#if !defined(USE_ROCM) && !defined(_MSC_VER)
```
Is this logic correct?  You want to disable this code for USE_ROCM && _MSC_VER because HIP on Windows doesn't support sparse.  But all other prior cases should remain.  Meaning, ROCm on linux, CUDA on linux and windows.
```suggestion
#if !(defined(USE_ROCM) && defined(_MSC_VER))
```
Or
```
#if !defined(USE_ROCM) || !defined(_MSC_VER)
```

ikalinic replied:
Thanks for the review and suggestion. Yes, we want to disable for just ROCm on Windows for now.

jeffdaily reviewed (changes requested):
Double-check your #if cases elsewhere to make sure you aren't causing a side effect of disabling other platforms besides rocm+windows.

ikalinic committed:
Fix logic for disabling hipSPARSE and ck declarations

Makes disabling hipSPARSE and ck declarations applicable to only ROCm on
Windows.

jeffdaily committed:
reduce how many ifdef's we need

jeffdaily commented:
@pytorchbot merge
